test/unit/options_validator_tests.js

```diff
@@ -494,4 +494,76 @@ describe('Options Validation', function() {
       testClass.testOperation({ a: 1 }, { b: 2 }, () => {}, () => {});
     }).to.throw(errorMessage);
   });
+
+  it('Should correctly handle a promise', function() {
+    class TestClass {
+      constructor() {
+        this.s = { options: { validationLevel: 'error' } };
+      }
+    }
+
+    TestClass.prototype.testOperation = arityOne()
+      .options({ b: { type: 'boolean', default: false } })
+      .build(function(requiredArgument, options, callback) {
+        return this.testOperation2(requiredArgument, options, callback);
```

Comment:
This is an interesting case, b/c ideally we do not want something to pass through options management more than once. Is this showing up somewhere in the actual code base?

Reply:
Yes, `insert` calls `insertMany`, which would cause the options passed to `insert` to go through options management twice.

Reply:
Ah. Ideally, we'd rework `insert` to call the `insertMany` operation instead. Still, I think it makes sense to add these checks.

Reply:
Yeah, that makes sense. I can make a ticket to rework `insert`.

Reply:
👍 to all of this 😄